guardrails/guard.py

```diff
@@ -787,7 +826,15 @@ def __call__(
         """
         instructions = instructions or self._exec_opts.instructions
         prompt = prompt or self._exec_opts.prompt
-        msg_history = msg_history or []
+        msg_history = msg_history or self._exec_opts.messages or []
+        if prompt is not None or instructions is not None or len(msg_history) > 0:
+            warnings.warn(
+                "prompt, instructions, and msg_history are deprecated "
+                "Please use the 'messages' argument instead"
```

Comment:
the problem with this suggestion is that msg_history and in turn messages is not well supported for things like reask 🤔

Reply:
What makes messages not well supported for reask? Is it the ambiguity of which message to patch? If that's the case, I think we can just assume that we patch the last one.

Reply:
i think we clarified this yesterday in the bug bash but msg_history reask instruction overrides were never implemented so its not like a 1:1 api. it still does fall through to the prompt and instruction based reasking.